e2e/e2e.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
```

Comment:
I think the tests should stop on first error, and throw a recap.

you should probably:
- set -o errexit,pipefail 
- use a trap (see comment below)
- add the "recap" of the tests who succeeded/failed in the trap.

After that (and the indentation issues), I think we can merge up this script. In the future, I think it might be more robust to use a different test framework than a self made bash script. (Either a test framework in bash, or a golang test).